.github/workflows/update-schema-diagram.yaml

```diff
@@ -0,0 +1,54 @@
+name: Update Schema Diagram
+
+on:
+  workflow_call:
+    inputs:
+      python_version:
+        required: true
+        type: string
+
+jobs:
+  update-schema:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ github.token }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python_version }}
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y graphviz graphviz-dev
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install poetry
+          # Install pygraphviz with explicit include and library paths
+          CFLAGS="-I/usr/include/graphviz" LDFLAGS="-L/usr/lib/graphviz" poetry run pip install --no-cache-dir pygraphviz
+          poetry install
+
+      - name: Generate schema diagram
+        run: |
+          poetry run python scripts/generate_schema_diagram.py
+
+      - name: Check for changes
+        id: git-check
+        run: |
+          git add docs/diagrams/database-schema.png docs/diagrams/database-schema.puml
+          git diff --staged --quiet || echo "changes=true" >> $GITHUB_OUTPUT
+
+      - name: Commit changes
```

Comment:
I find having CI to autogenerate commits on my branch to be incredibly annoying, because I often make changes, rebase, and force push back to branches I'm working on, and these kinds of things get in the way of that workflow, particularly if they add a commit on every push.

I'm wondering if we can eliminate the need for committing generated files back to the repo. For example, perhaps when we have dev environments/review branches we could serve the images from the app itself, or perhaps we could publish documentation to github pages, and include the diagrams in that. ([here's an example from my previous team](https://github.com/ministryofjustice/find-moj-data-runbooks))

An equivalent but approach might be to move this workflow into a git hook via pre-commit.